python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -528,3 +507,36 @@ def _log_op_metrics(topology: Topology) -> None:
     for op in topology:
         log_str += f"{op.name}: {op.metrics.as_dict()}\n"
     logger.debug(log_str)
+
+
+class _ClosingIterator(OutputIterator):
```

Comment:
nit, I still prefer the old name `StreamingIterator`, as it's the iterator for StreamingExecutor. 
"closing" is one of the implementation details.

Reply:
Streaming in `StreamingExecutor` makes sense, but iterator itself is pretty ordinary implementing standard iterator's protocol (ie there's nothing really streaming about the iterator itself).